### PR TITLE
Make lesson by random words

### PR DESCRIPTION
**Pull request by thiennv57**

### CODE REVIEW

File: app/assets/stylesheets/custom.css.scss [hunk 1]
```diff
@@ -52,4 +52,7 @@ display: inline;
 background-color: #000;
 width: 1px;
 height: 100%;
```
Comment (ThuBM) [inline app/assets/stylesheets/custom.css.scss]: indent lai file nay!

File: app/models/word.rb [hunk 2]
```diff
@@ -12,4 +12,5 @@ class Word < ActiveRecord::Base
   scope :not_learned, ->user{where("id NOT IN (
     SELECT word_id FROM results WHERE lesson_id IN ( SELECT id FROM lessons WHERE user_id = ?
       ))",user.id)}
+  scope :random_words, ->user {not_learned(user).limit(Settings.question_limit_words).order "RANDOM()"}
```
Comment (ThuBM) [inline app/models/word.rb]: Em viet lai cai scope nay nhe'. ko nen viet scope nay goi trong scope kia nhu the nay!

File: app/views/lessons/_question.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<div class="result">
+  <hr>
+  <div class="question">
+    <div class="col-md-5">
+      <%= t"question" %><%= f.index + Settings.question.add_index %>
```
Comment (ThuBM) [inline app/views/lessons/_question.html.erb]: Them space vao giua   t "question"

File: app/views/lessons/_question.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      <br>
+      <h1><%= f.object.word.word %></h1>
+    </div>
+    <div class="col-md-5">
+      <%= f.hidden_field :word_id, value: f.object.word.id %>
+      <% f.object.word.answers.each do |answer| %>
+        <div class="row">
+          <div class="col-md-2">
+          <%= f.radio_button :answer_id, answer.id %></div>
```
Comment (ThuBM) [inline app/views/lessons/_question.html.erb]: Indent

File: app/models/lesson.rb [hunk 5]
```diff
@@ -2,4 +2,13 @@ class Lesson < ActiveRecord::Base
   belongs_to :category
   belongs_to :user
   has_many :results, dependent: :destroy
+  accepts_nested_attributes_for :results, allow_destroy: true
```
Comment (ThuBM) [inline app/models/lesson.rb]: allow_destroy false

File: app/models/lesson.rb [hunk 6]
```diff
@@ -2,4 +2,13 @@ class Lesson < ActiveRecord::Base
...
+  
+  def num_of_correct_ans
```
Comment (ThuBM) [inline app/models/lesson.rb]: method nay duoc su dung o dau?

File: app/models/lesson.rb [hunk 7]
```diff
@@ -2,4 +2,13 @@ class Lesson < ActiveRecord::Base
...
+    count = 0
+    self.results.each do |result|
+       count += 1 if result.answer.is_correct
+    end
+    count
+  end
```
Comment (ThuBM) [inline app/models/lesson.rb]: Em viet lai ham nay, tinh count bang scope

File: app/models/lesson.rb [hunk 8]
```diff
@@ -2,4 +2,6 @@ class Lesson < ActiveRecord::Base
   belongs_to :category
   belongs_to :user
   has_many :results, dependent: :destroy
+  accepts_nested_attributes_for :results, allow_destroy: false
```
Comment (ThuBM) [inline app/models/lesson.rb]: remove allow_destroy:false

File: app/models/result.rb [hunk 9]
```diff
@@ -1,4 +1,6 @@
 class Result < ActiveRecord::Base
   belongs_to :lesson
   belongs_to :word
+  has_one :result
```
Comment (ThuBM) [inline app/models/result.rb]: sao lai has_one :result ???

File: app/views/lessons/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,19 @@
+<% provide :title, t("lesson") %>
+<h1 align="center"><%= t "lesson" %></h1>
+<h3 align="center"><%= @category.name %></h3>
+<div class="container">
+  <div class="row" id="lesson">
+    <%= form_for [@category, @lesson] do |f| %>
+      <%= hidden_field_tag :category_id, @category.id %>
```
Comment (chuanhtuan) [inline app/views/lessons/new.html.erb]: f.hidden_field

Comment (chuanhtuan) [inline app/views/lessons/new.html.erb]: remove

**Pull request merged**